### ISSUE TITLE
Issue with Abode Login

### ISSUE THREAD

**Issue by clknight2111** [chrisbsmith/homebridge-abode-lights#218]
Versions 1.03 and later cause the following error:
```
[29/11/2022, 06:21:27] [Abode Lights] Failed to performAuth: Response did not contain API key.
[29/11/2022, 06:21:27] [Abode Lights] Failed to initialize: Failed to sign into Abode account
```
Rolling back to 1.02 fixes the problem. 

FWIW, I have 2FA enabled on my primary Abode account but not on the separate account I use for homebridge login.

Thanks!
-Chris

**Comment by chrisbsmith** [comment]
Thanks for sending this along @clknight2111 .

I just set up a fresh homebridge (using docker) and installed `1.0.5` and didn't have the same login issue.

```
[11/29/2022, 2:11:51 PM] [Abode Lights] Initializing Abode Lights platform...
[11/29/2022, 2:11:51 PM] [Abode Lights] Signing into Abode account
```

Was it working for you on 1.02 and then upgrading to 1.03 failed? Can you try upgrading to 1.0.5 and see if it happens again?

**Comment by clknight2111** [comment]
The issue emerged when I upgraded from 1.0.2 -> 1.0.5; I went back through the versions until it worked again when I went from 1.0.3 to 1.0.2. Can’t think of anything unique in my Abode setup but I’m happy to send logs if there’s a way to generate them. Thanks!—- ChrisOn Nov 29, 2022, at 11:16 AM, Chris Smith ***@***.***> wrote:﻿
Thanks for sending this along @clknight2111 .
I just set up a fresh homebridge (using docker) and installed 1.0.5 and didn't have the same login issue.
[11/29/2022, 2:11:51 PM] [Abode Lights] Initializing Abode Lights platform...
[11/29/2022, 2:11:51 PM] [Abode Lights] Signing into Abode account

Was it working for you on 1.02 and then upgrading to 1.03 failed? Can you try upgrading to 1.0.5 and see if it happens again?

**Comment by chrisbsmith** [comment]
@clknight2111 I was able to reproduce this error this morning. I'll look into this as soon as I can.

**Comment by brentopolis** [comment]
I’m seeing this same error after updating to 1.0.5  today.  Unfortunately, I’m not sure what version I was on previously.  I tried uninstalling and reinstalling the plugin, which didn’t work.  Rolling back to 1.0.2 did work.

**Comment by chrisbsmith** [comment]
@brentopolis And @clknight2111, I've got a few questions for you to help me troubleshoot: 
- how are you running Homebridge?
- What version of Homebridge are you running?
- have you tried uninstalling an reinstalling the plugging? (Sorry I know this will kill any HomeKit automations you have set up)
- do you have the same auth issues with 1.0.3 and 1.0.4?

**Comment by brentopolis** [comment]
Hi @chrisbsmith unfortunately I’m traveling for work this week and I don’t have my homebridge setup exposed to the internet, so I won’t be able to test until Friday. 

As to your questions:
* I’m running on a raspberry pi, installed via npm
* I’m not sure offhand, but I updated everything yesterday (Dec 4) so it should be the latest. I updated homebridge itself and a few plugins. 
* I did try uninstalling and reinstalling, but it didn’t help 
* I didn’t try 1.0.3 or 1.0.4 — I went straight to 1.0.2 based on @clknight2111 ‘s experience

**Comment by chrisbsmith** [comment]
No worries. I am fairly confident the issue is with the axios version bump in 1.0.3. 

I'll wait to hear from you and then will cut a 1.0.6 patch.

**Comment by clknight2111** [comment]
This alpha release fixed it for me! Thanks!—- ChrisOn Dec 5, 2022, at 11:54 AM, Chris Smith ***@***.***> wrote:﻿
@brentopolis and @clknight2111 , I am working on a fix for this and have published version 1.0.6-alpha.0. I downgraded the version of axios to the a pre 1.X version and it seems to have resolved the issues in my dev testing.
Can you install this version and see if authentication works again? It is available as an alt version.

If this works out, I'll remove the alpha and push a new release.

**Comment by chrisbsmith** [comment]
Thanks for the feedback. I'll release `1.0.6` today.
